proposals/lip-state-model.md

```diff
@@ -0,0 +1,103 @@
+```
+LIP: <LIP number>
+Title: State model and state root
+Author: Alessandro Ricottone <alessandro.ricottone@lightcurve.io>
+Type: Informational
+Created: <YYYY-MM-DD>
+Updated: <YYYY-MM-DD>
+Required: 0039
+```
+
+
+## Abstract
+
+The state of an interoperable chain in the Lisk ecosystem is organized as a sparse Merkle tree, the state tree, built on top of a generic key-value storage for each module of the chain.
+This allows to authenticate the whole state with the tree Merkle root, the state root.
+In this LIP, we define the state model of a Lisk blockchain and the construction of the state tree from which the state root is calculated.
+
+
+## Copyright
+
+This LIP is licensed under the [Creative Commons Zero 1.0 Universal](https://creativecommons.org/publicdomain/zero/1.0/).
+
+
+## Motivation
+
+A blockchain created with the  Lisk SDK is organized in a modular way: each module registered in the chain defines its own state and the possible state transitions, i.e., the transactions defined within the module or the handlers that can be called by other modules.
+This is reflected by the state tree: each module maintains its own key-value storage and sparse Merkle tree, which are finally merged together to form the complete state tree.
+This LIP specifies the general format for the key-value storages and the way in which they are combined together in the state tree. 
+
+Organizing the state of a blockchain as a Merkle tree allows to cryptographically authenticate the whole state with a single hash, the state root.
+The state root property is calculated at the end of the block processing and included in the block header.
+When a certificate is posted to a chain, the state root is used to authenticate the information contained in it against the block header, which has been signed by the sending chain validators.
```

Comment:
Actually, the certificate and not the block header is signed by the validators.

Reply:
edited substantially